src/System.Threading.Timer/tests/TimerFiringTests.cs

```diff
@@ -0,0 +1,226 @@
+        {
+            Assert.False(are.WaitOne(TimeSpan.FromSeconds(1)));
+        }
+    }
+
+    [Fact]
+    public void MultipleTimers_ShouldFire_InCorrectOrder()
+    {
+        Object t1State = new Object();
+        Object t2State = new Object();
+        Timer t1 = null;
+        Timer t2 = null;
+        int count = 0;
+        AutoResetEvent are = new AutoResetEvent(false);
+        TimerCallback tc = new TimerCallback((object o) =>
+        {
+            if (count == 0)
+            {
+                Assert.Same(t2State, o);
+                count++;
+            }
+            else
+            {
+                Assert.Same(t1State, o);
+                are.Set();
+            }
+        });
+
+        using (t1 = new Timer(tc, t1State, TimeSpan.FromSeconds(1), TimeSpan.FromMilliseconds(-1)))
+        using (t2 = new Timer(tc, t2State, TimeSpan.FromMilliseconds(0), TimeSpan.FromMilliseconds(-1)))
+        {
+            // Wait for both events to fire
+            Assert.True(are.WaitOne(TimeSpan.FromMilliseconds(1500 /*1.5 seconds*/)));
+        }
+    }
+
+    [Fact]
+    public void Timer_CanDisposeSelfInCallback()
+    {
+        // There's nothing to validate besides that we don't throw an exception, so rely on xunit 
+        // to catch any exception that would be thrown and signal a test failure
+        Timer t = null;
+        TimerCallback tc = new TimerCallback((object o) => t.Dispose());
+        t = new Timer(tc, null, 100, -1);
+    }
+
+    [Fact]
+    public void Timer_CanBeDisposedMultipleTimes()
+    {
+        // There's nothing to validate besides that we don't throw an exception, so rely on xunit 
+        // to catch any exception that would be thrown and signal a test failure
+        TimerCallback tc = new TimerCallback((object o) => { });
+        var t = new Timer(tc, null, 100, -1);
+        for (int i = 0; i < 10; i++)
+            t.Dispose();
+    }
+
+    [Fact]
+    public void NonRepeatingTimer_ThatHasAlreadyFired_CanChangeAndFireAgain()
+    {
+        AutoResetEvent are = new AutoResetEvent(false);
+        TimerCallback tc = new TimerCallback((object o) => are.Set());
+        using (var t = new Timer(tc, null, 1, -1))
+        {
+            Assert.True(are.WaitOne(50), "Should have received first timer event");
+            Assert.False(are.WaitOne(20), "Should not have received a second timer event");
+            t.Change(1, -1);
+            Assert.True(are.WaitOne(50), "Should have received a second timer event after changing it");
+        }
+    }
+
+    [Fact, OuterLoop]
+    public void Running_Timer_CanBeDisposedAndStopsFiring()
+    {
+        AutoResetEvent are = new AutoResetEvent(false);
+        TimerCallback tc = new TimerCallback((object o) => are.Set());
+        var t = new Timer(tc, null, 100, 500);
+        Assert.True(are.WaitOne(250), "Failed to get first timer fire");
+        t = null; // Remove our refence so the timer can be GC'd 
+        GC.Collect();
```

Comment:
Does this test pass locally for you consistently?  I'd have expected this could need:
```C#
GC.Collect();
GC.WaitForPendingFinalizers();
GC.Collect();
```
You ran the outer loop tests locally?